src/backend/revoker/indexer/queryFromIndexer.test.ts

```diff
@@ -0,0 +1,297 @@
+      it("should only query once if all matches are on the first Indexer's response", async () => {
+        const count = Math.floor(QUERY_SIZE * 0.77);
+
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: count,
+              nodes: mockBlocks(count).map((b) => ({ ...b })),
+            },
+          },
+        };
+        const buildBlockQuery = buildBlockQueries();
+        const aFewMatches = matchesGenerator<IndexedBlock>(buildBlockQuery);
+
+        for await (const match of aFewMatches) {
+          expect(match).toBeDefined();
+        }
+
+        expect(got.post).toHaveBeenCalledTimes(1);
+      });
+      it('should continue requesting from the Indexer until querying all matches', async () => {
+        const count = Math.floor(QUERY_SIZE * 3.33);
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: count,
+              nodes: mockBlocks(QUERY_SIZE).map((b) => ({ ...b })),
+            },
+          },
+        };
+
+        const buildBlockQuery = buildBlockQueries();
+        const aLotOfMatches = matchesGenerator<IndexedBlock>(buildBlockQuery);
+
+        for await (const match of aLotOfMatches) {
+          expect(match).toBeDefined();
+        }
+
+        expect(got.post).toHaveBeenCalledTimes(5);
+
+        const { calls: postRequests } = jest.mocked(got.post).mock;
+
+        // @ts-expect-error because TS infers wrong parameters
+        expect(postRequests[0][1]).toMatchObject({
+          json: { query: buildBlockQuery(0) },
+        });
+        // @ts-expect-error because TS infers wrong parameters
+        expect(postRequests[1][1]).toMatchObject({
+          json: { query: buildBlockQuery(0) },
+        });
+        // @ts-expect-error because TS infers wrong parameters
+        expect(postRequests[2][1]).toMatchObject({
+          json: { query: buildBlockQuery(QUERY_SIZE) },
+        });
+        // @ts-expect-error because TS infers wrong parameters
+        expect(postRequests[3][1]).toMatchObject({
+          json: { query: buildBlockQuery(QUERY_SIZE * 2) },
+        });
+        // @ts-expect-error because TS infers wrong parameters
+        expect(postRequests[4][1]).toMatchObject({
+          json: { query: buildBlockQuery(QUERY_SIZE * 3) },
+        });
+      }, 10000);
+    });
+    describe('on negative cases', () => {
+      it('should just return void if the GraphQL Endpoint is not available', async () => {
+        configuration.indexer.graphqlEndpoint = 'NONE';
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: 37,
+              nodes: mockBlocks(37).map((b) => ({ ...b })),
+            },
+          },
+        };
+
+        const noMatches = matchesGenerator<IndexedBlock>((foo) =>
+          foo.toString(),
+        );
+
+        const returnValue = await noMatches.next();
+        expect(returnValue.value).toBeUndefined();
+        expect(returnValue.done).toBe(true);
```

Comment:
```suggestion
        for await (const unexpected of noMatches) {
          throw new Error('Generator unexpectedly yielded a value')
        }
```
While technically correct, this usage of API is different from what we do in the code. Would it be more clear if we aligned them?

Reply:
I thought that throwing on tests is normally avoided.

Reply:
I might be wrong, but this is the first time I hear about this approach. I am usually very practical about my tests, and doing something that makes it fail is usually fine by me. Like throwing an exception or doing something that will throw an exception.

Reply:
I just induced that rule from watching most test. I guess it is not an established rule.

Reply:
Can I still keep the version with `.next()`?

Reply:
All right